pkg/gateway/operations/putobject.go

```diff
@@ -298,6 +298,15 @@ func handlePut(w http.ResponseWriter, req *http.Request, o *PathOperation) {
 	o.Incr("put_object", o.Principal, o.Repository.Name, o.Reference)
 	storageClass := StorageClassFromHeader(req.Header)
 	opts := block.PutOpts{StorageClass: storageClass}
+	allowOverWrite, err := o.checkIfAbsent(req)
```

Comment:
For future generations, please add comment explaining this, something or exactly like [this](https://github.com/treeverse/lakeFS/blob/544e95e5848ea69a4700cfebfe06ff34973e6b9e/pkg/api/controller.go#L3170) otherwise it's confusing, since this does not actually the verify core part, it's more optimization (since not atomicity here)